optuna/distributions.py

```diff
@@ -233,6 +316,61 @@ def _contains(self, param_value_in_internal_repr: float) -> bool:
         return self.low <= value <= self.high and abs(k - round(k)) < 1.0e-8
 
 
+class IntDistribution(BaseDistribution):
+    """A discrete uniform distribution on integers.
+
+    This object is instantiated by :func:`~optuna.trial.Trial.suggest_float`, and passed to
+    :mod:`~optuna.samplers` in general.
+
+    Attributes:
+        low:
+            Lower endpoint of the range of the distribution. ``low`` is included in the range.
+        high:
+            Upper endpoint of the range of the distribution. ``high`` is included from the range.
+        log:
+            If ``log`` is True, this distribution is in log-scaled domain.
+
+    Raises:
+        ValueError:
+            If ``low`` value is larger than ``high`` value,
+            or ``low`` value is less than 1 when ``log`` is :obj:`True.
+    """
+
+    def __init__(self, low: int, high: int, log: bool = False) -> None:
+        if low > high:
+            raise ValueError(
+                "The `low` value must be smaller than or equal to the `high` value "
+                "(low={}, high={}).".format(low, high)
+            )
+
+        if log and low < 1:
+            raise ValueError(
+                "The `low` value must be equal to or greater than 1 for a log distribution "
+                "(low={}, high={}).".format(low, high)
+            )
+
+        self.low = low
+        self.high = high
+        self.log = log
```

Comment:
Better to cast them into `int`.

Reply:
Thank you for prompt feedbacks! I have casted before passing them to `_adjust_int_uniform_high()`. 
Please kindly let me know if any concern.